### PR TITLE
fix(llm): 🐛 fix long Memo Tag issue on Cardano

### PR DESCRIPTION
**Pull request by thesan**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Truncate Cardano memos in the input to prevent the transaction validation errors.
(Follow up to #8340 and #8604).

https://github.com/user-attachments/assets/cb83d90d-281a-4afb-8967-b645a50b7e2d

### ❓ Context

- [LIVE-15161]
- [LIVE-15162]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-15161]: https://ledgerhq.atlassian.net/browse/LIVE-15161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-15162]: https://ledgerhq.atlassian.net/browse/LIVE-15162

### CODE REVIEW

**Comment by themooneer** [comment]
i've this issue when i paste a cardano address 
image

File: apps/ledger-live-mobile/src/families/cardano/MemoTagInput.tsx [hunk 1]
```diff
@@ -3,10 +3,17 @@ import React from "react";
 import type { Transaction as CardanoTransaction } from "@ledgerhq/live-common/families/cardano/types";
 import type { MemoTagInputProps } from "LLM/features/MemoTag/types";
 import { GenericMemoTagInput } from "LLM/features/MemoTag/components/GenericMemoTagInput";
+import { truncateUtf8 } from "LLM/utils/truncateUtf8";
 
 export default (props: MemoTagInputProps<CardanoTransaction>) => (
   <GenericMemoTagInput
     {...props}
```
Comment (themooneer) [inline apps/ledger-live-mobile/src/families/cardano/MemoTagInput.tsx:10]: you can add 
```typescript
maxLength={MAX_MEMO_LENGTH}
```

**Review by themooneer — APPROVED** [review]

**Review by ComradeAERGO — APPROVED** [review]

**Comment by thesan** [comment]
TLDR I removed the last commit because I did one last test which went wrong and there's a minor bug on the backend.

If I send a memo with loads of emojies like 50👍 then `maxLength` will cut it to 32👍. It felt wrong to just use 2bytes per 👍 but it passes the backend doesn't report the "string or buffer length invalid" (like in [LIVE-15162](https://ledgerhq.atlassian.net/browse/LIVE-15162)) so I assumed it was fine.
However when I actually sent the transaction it failed with the backend returning: `API HTTP 400 https://cardano.coin.ledger.com/api/v1/transaction/submit`. When reverting to `truncateUtf8` it correctly truncate to 16👍 (which should fit in 64bytes).

This means that the backend validation is incorect it probably checks for the length of the string instead of the bytes size of the string (in most emojies get encoded in 2 characters).

**Review by themooneer — APPROVED** [review]
